Sources/FoundationEssentials/Date.swift

```diff
@@ -239,20 +239,31 @@ extension Date : CustomDebugStringConvertible, CustomStringConvertible, CustomRe
     /// function `description(locale:)`.
     public var description: String {
         // NSDate uses the constant format `uuuu-MM-dd HH:mm:ss '+0000'`
-        let outputLength = 26 // Constant length of `uuuu-MM-dd HH:mm:ss '+0000'` + \0
         let format = "%Y-%m-%d %H:%M:%S +0000"
+        let unavailable = "<description unavailable>"
+
         var info = tm()
         var time = time_t(self.timeIntervalSince1970)
         gmtime_r(&time, &info)
-        let result = UnsafeMutablePointer<CChar>.allocate(capacity: outputLength)
-        defer {
-            result.deallocate()
-        }
-        guard strftime(result, outputLength, format, &info) >= 0 else {
-            return "<description unavailable>"
-        }
 
-        return String(validatingUTF8: result) ?? "<description unavailable>"
+        // This allocates stack space for range of 10^102 years
+        // That's more than Date currently supports.
+        let bufferSize = 128
```

Comment:
`struct tm` stores its year with an Int. That means that the farthest date we care about is the last moment of year `Int64.max`. Considering that a 128 buffer is probably overkill?

Reply:
You also mentioned this in latest commit 

> * That limitation leads to (currently) the result never using more than 26 bytes … which makes the change to temporary allocation no longer required and 128 bytes even more overkill. For now, I kept it.

Why don't you just make this change while you're here?

Reply:
I don’t think it really matters wether we allocate 26 bytes, 32, 64, or 128 bytes.

I don’t want to use 26 bytes because that will break when .distantFuture/Past is extended. If these limits get changed, 32 is also well within reach. So what would I change it to? A good value would be whatever TimeInterval allows. I’m not sure what this would be, probably somewhere around 330. Not super alignment friendly so I’d use 384 then.

64 would be enough, even for +/-Int128 but then, again, I don’t think it does matter.

What I _really_ would like to do is something like `"\(tm.tm_year, minWidth: 4, padding: "0")-\(tm.tm_month+1, width: 2, padding: "0")-\(tm. tm_mday, width: 2, padding: "0") \(tm.tm_hour, width: 2, padding: "0"):\(tm.tm_min, width: 2, padding: "0"):\(tm.tm_sec, width: 2, padding: "0")"` because that would remove the overhead of parsing a (constant) template, the overhead of validating UTF-8, and problems with non-standard `strftime` (I cannot test the WinSDK path).

Reply:
Just in case I failed to transport it in English: If you think this should be a different value, I’ll change it.

I just think that 26 is not future prove (and given the `%4Y` thing, I do no longer trust `strftime` implementations to not output a bit more than reasonable). Given that, I opted to be on the safe (safer) side by using 128 instead of 64. If stack space is a problem, I’d use 32 (alignment friendly version of 26 with a few bytes buffer).

Reply:
What do you mean by a non-standard `strftime`?  The windows version should be conformant as per the spec.

Reply:
I don't feel strongly towards any other number. It's just that I thought you agreed that 128 was overkill, so I was curious why you didn't change that number to something less likely being an overkill. I'm ok with this